packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,599 @@
 
 part of firebase_ml_vision;
 
+/// Detector for performing optical character recognition(OCR) on an input image.
+///
+/// A barcode detector is created via getVisionBarcodeDetector() in [FirebaseVision]:
+///
+/// ```dart
+/// BarcodeDetector barcodeDetector = FirebaseVision.instance.barcodeDetector();
+/// ```
 class BarcodeDetector extends FirebaseVisionDetector {
-  BarcodeDetector._(BarcodeDetectorOptions options);
+  BarcodeDetector._();
 
+  /// Closes the barcode detector and release its model resources.
   @override
   Future<void> close() async {
-    // TODO: implement close
+    return FirebaseVision.channel.invokeMethod('BarcodeDetector#close');
   }
 
+  /// Detects barcode in the input image.
+  ///
+  /// The OCR is performed asynchronously.
   @override
-  Future<void> detectInImage(FirebaseVisionImage visionImage) async {
-    // TODO: implement detectInImage
+  Future<List<VisionBarcode>> detectInImage(
+      FirebaseVisionImage visionImage) async {
+    final List<dynamic> reply = await FirebaseVision.channel.invokeMethod(
+      'BarcodeDetector#detectInImage',
+      visionImage.imageFile.path,
+    );
+
+    final List<VisionBarcode> barcodes = <VisionBarcode>[];
+    reply.forEach((dynamic barcode) {
+      barcodes.add(new VisionBarcode._(barcode));
+    });
+
+    return barcodes;
   }
 }
 
-class BarcodeDetectorOptions {}
+/// Represents a single recognized barcode and its value.
+class VisionBarcode {
```

Comment:
I think using `Barcode` rather than `VisionBarcode` would be sufficient. It would also match the other Detectors. 

I think you can remove `Vision` from all the other classes as well, since it makes the class names longer, but does not help in terms of clarity.

Reply:
I was wondering what to do about it! I fix it😁